src/main/java/net/sf/jabref/importer/fetcher/ISBNtoBibTeXFetcher.java

```diff
@@ -61,14 +62,26 @@ public void stopFetching() {
 
     @Override
     public boolean processQuery(String query, ImportInspector inspector, OutputPrinter status) {
+        Optional<BibEntry> entry = getEntryFromISBN(query, status);
+        if (entry.isPresent()) {
+            inspector.addEntry(entry.get());
+            return true;
+        }
+        return false;
+
+    }
+
+    public Optional<BibEntry> getEntryFromISBN(String query, OutputPrinter status) {
```

Comment:
Could you please implement the new `IdBasedFetcher` interface (i.e. throw a `FetcherException` whenever the output printer was used). I think this makes in particular sense since you pass `null` as the second argument above (which would result in a NPE in case of an error)